File: credit/credit.c
#include <cs50.h>
#include <stdio.h>
#include <stdbool.h>

// Function prototypr slot

bool validity_checker(long long ccnum);
int find_length_cc(long long ccnum);
bool checksum(long long ccnum);
void print_credit_card_name(long long ccnum);

int main(void)
{
    long long ccnum;
// asking for credit card nummber
    do
    {
        ccnum = get_long("Enter your credit card number: \n");
    }
    while (ccnum <= 0);
// checking credit card if valid or invalid
    if (validity_checker(ccnum))
    {
        print_credit_card_name(ccnum);
    }
    else
    {
        printf("INVALID\n");
    }
}

// creating function bodies
// Comparing length to size credit card number
bool validity_checker(long long ccnum)
{
    int len = find_length_cc(ccnum);
    return ((len == 13 || len == 15 || len == 16) && checksum(ccnum));
}
// Checking length of credit card number
int find_length_cc(long long ccnum)
{
    int len = 0;
    while (ccnum != 0)
    {
        ccnum = ccnum / 10;
        len++;
    }
    return len;
}
// Getting the last digit credit card number
bool checksum(long long ccnum)
{
    int sum = 0;
    long long tempccnum = ccnum;
    while (tempccnum > 0)
    {
        int lastdigit = tempccnum % 10;
        sum = sum + lastdigit;
        tempccnum = tempccnum / 100;
    }
// Getting secondlast digit credit card number
// Calculating every other number plus multiply by 2 and adding remainder numbers
    tempccnum = ccnum / 10;
    while (tempccnum > 0)
    {
        int secondlastdigit = (tempccnum % 10);
        int tempmultiply = secondlastdigit * 2;
        sum = sum + (tempmultiply % 10) + (tempmultiply / 10);
        tempccnum = tempccnum / 100;
    }
    return (sum % 10) == 0;
}
// printing credit card name
void print_credit_card_name(long long ccnum)
{
    if ((ccnum >= 34e13 && ccnum < 35e13) || (ccnum >= 37e13 && ccnum < 38e13))
    {
        printf("AMEX\n");
    }
    else if (ccnum >= 51e14 && ccnum < 56e14)
    {
        printf("MASTERCARD\n");
    }
    else if ((ccnum >= 4e12 && ccnum < 5e12) || (ccnum >= 4e15 && ccnum < 5e15))
    {
        printf("VISA\n");
    }
    else
    {
        printf("INVALID\n");
    }
}

File: Python/sentimental-credit/credit.py
from cs50 import get_int

# Main function


def main():
    while True:
        # Getting number from user
        credit_card = get_int("Number: ")
        if credit_card >= 0:
            break
# checking card number
    if validity_checker(credit_card):
        # Printing name of card
        print_credit_card_name(credit_card)
    else:
        print("INVALID")


def validity_checker(ccnum):
    # Checking which card it is and if valid
    return checksum(ccnum)

# luhn algorithm


def checksum(ccnum):
    sum = 0
    for i in range(len(str(ccnum))):
        if (i % 2 == 0):
            sum += ccnum % 10
        else:
            digit = 2 * (ccnum % 10)
            sum += digit // 10 + digit % 10

        ccnum //= 10

    return sum % 10 == 0

# Printing name or INVALID


def print_credit_card_name(ccnum):
    if (ccnum >= 34e13 and ccnum < 35e13) or (ccnum >= 37e13 and ccnum < 38e13):
        print("AMEX")

    elif (ccnum >= 51e14 and ccnum < 56e14):
        print("MASTERCARD")

    elif (ccnum >= 4e12 and ccnum < 5e12) or (ccnum >= 4e15 and ccnum < 5e15):
        print("VISA")

    else:
        print("INVALID")


# Main function
if __name__ == "__main__":
    main()